lib/eco/export/jruby.py

```diff
@@ -365,49 +367,74 @@ def annotate_tree(self, src_file_name, log_file_name):
 
         # Process graph of reachable objects and annotate parse tree.
         self.remove_all_annotations()
+
         for obj in reachable_objects:
-            if (isinstance(obj, Method) and obj.source.file != "(unknown)"
-                and not obj.is_hidden() and not obj.is_core()
-                and not obj.is_library() and obj.source.file != "(eval)"
-                and obj.name != "<main>"):
+            if (isinstance(obj, Method) and obj.source.file != "(unknown)" and
```

Comment:
AIUI the "(unknown)" is inserted by the interpreter when a method has been automatically generated, and not written into the code by the programmer. So, we are safe until the callgraph log changes its syntax, in which case we will no doubt have to make further changes anyway.